qiskit/aqua/algorithms/amplitude_estimators/ae.py

```diff
@@ -444,4 +446,151 @@ def _run(self) -> dict:
         kind = 'likelihood_ratio'  # empirically the most precise kind
         self._ret['95%_confidence_interval'] = self.confidence_interval(alpha, kind)
 
-        return self._ret
+        ae_result = AmplitudeEstimationAlgorithmResult()
+        ae_result.value = self._ret['value']
+        ae_result.estimation = self._ret['estimation']
+        ae_result.num_oracle_queries = self._ret['num_oracle_queries']
+        ae_result.confidence_interval = self._ret['95%_confidence_interval']
+
+        result = AmplitudeEstimationResult()
+        result.combine(ae_result)
+        result.ml_value = self._ret['ml_value']
+        result.ret_values = self._ret['values']
+        result.probabilities = self._ret['probabilities']
+        result.shots = self._ret['shots']
+        result.mle = self._ret['mle']
+        if 'statevector' in self._ret:
+            result.statevector = self._ret['statevector']
+        if 'counts' in self._ret:
+            result.counts = self._ret['counts']
+        result.a_items = self._ret['a_items']
+        result.y_items = self._ret['y_items']
+        result.mapped_values = self._ret['mapped_values']
+        result.max_probability = self._ret['max_probability']
```

Comment:
@Cryoris The 'from_dict' expects a dictionary input with the same properties as the Result class. Since Some properties above change names, like 'values' etc, unless a new temp.dictionary were created with the correct properties names it wouldn't work.

Reply:
Ah right, thanks for explanation!